src/frontend/components/DashboardVideoLiveControlPane/widgets/DashboardVideoLiveWidgetVOD/index.tsx

```diff
@@ -0,0 +1,106 @@
+import { Box, Paragraph, Spinner } from 'grommet';
+import React from 'react';
+import { defineMessages, useIntl } from 'react-intl';
+
+import { DashboardVideoLiveWidgetTemplate } from 'components/DashboardVideoLiveControlPane/widgets/DashboardVideoLiveWidgetTemplate';
+import { liveState, Video } from 'types/tracks';
+
+import { HarvestVOD } from './HarvestVOD';
+import { PublishVOD } from './PublishVOD';
+
+const messages = defineMessages({
+  info: {
+    defaultMessage: 'This widget allows you to handle live vod features.',
+    description: 'Info of the widget used for live vod.',
+    id: 'components.DashboardVideoLiveWidgetVOD.info',
+  },
+  title: {
+    defaultMessage: 'VOD',
+    description: 'Title of the widget used for live vod.',
+    id: 'components.DashboardVideoLiveWidgetVOD.title',
+  },
+  noHarvestMessage: {
+    defaultMessage:
+      'There is nothing to harvest. To create your video, you have to record your stream.{br}Harvest will be available at the end of the live.',
+    description:
+      'Message displays in the VOD widget when the is nothing to harvest.',
+    id: 'components.DashboardVideoLiveWidgetVOD.noHarvestMessage',
+  },
+  harvestingInProgress: {
+    defaultMessage: 'Harvesting in progress...',
+    description: 'Message displays when live is harvesting.',
+    id: 'components.DashboardVideoLiveWidgetVOD.harvestingInProgress',
+  },
+  publishVODMessage: {
+    defaultMessage:
+      'Your vod is now available, you can download it to control the content then publish it.',
+    description:
+      'Message displayed in the vod widget when live have been harvested and we can publish the vod',
+    id: 'components.DashboardVideoLiveWidgetVOD.publishVODMessage',
+  },
+  publishVODButtonLabel: {
+    defaultMessage: 'Publish the vod',
+    description: 'Button label to publish the vod',
+    id: 'components.DashboardVideoLiveWidgetVOD.publishVODButtonLabel',
+  },
+});
+
+interface DashboardVideoLiveWidgetVODProps {
+  video: Video;
+}
+
+export const DashboardVideoLiveWidgetVOD = ({
+  video,
+}: DashboardVideoLiveWidgetVODProps) => {
+  const intl = useIntl();
+
+  const displayMessageLayer =
+    (video.live_state !== liveState.STOPPED &&
+      video.live_state !== liveState.HARVESTED) ||
+    !video.recording_time ||
+    video.recording_time <= 0;
```

Comment:
```suggestion
    video.recording_time <= 0;
```
can it be inf to 0?

Reply:
it is not suppose to but this is a number